app/services/business_entity_service.rb

```diff
@@ -0,0 +1,56 @@
+class BusinessEntityService
```

Comment:
I think it's quite solid, if you look at @dotemacs 's suggestions bellow, it'll be more readable, I agree. 

The only extra suggestion I have is to change the name of `@business_entity` to something like `@existing_business_entity`. So that it's clear especially when we're saving both in transaction.